homeassistant/components/velbus/__init__.py

```diff
@@ -138,6 +140,10 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     await hass.data[DOMAIN][entry.entry_id]["cntrl"].stop()
     hass.data[DOMAIN].pop(entry.entry_id)
+    await hass.async_add_executor_job(
+        shutil.rmtree,
```

Comment:
I think we should only remove the cache when the config entry is removed, not on reload. Ie implement `async_remove_entry` instead. Does that make sense?

Reply:
whats the exact difference between unload and remove?

unload = disable the entry?
remove = delete the entry?

if this is the case then yes, the velbus cache removal should only be done in the remove callback.

if the entry is removed, will it forst cal the unlaod and then the remove? or should i duplicate some code?

Reply:
Unload is called both when reloading the config entry and when deleting the config entry.

Remove is only called when deleting the config entry.

Reply:
implemented in #81031